cl/opinion_page/decorators.py

```diff
@@ -0,0 +1,41 @@
+from functools import wraps
+
+from django.http import Http404
+from django.shortcuts import redirect
+
+from cl.search.models import ClusterRedirection
+
+
+def handle_cluster_redirection(view_func):
+    """
+    Redirect from deleted clusters to existing clusters
+
+    Uses the ClusterRedirection table, and only changed the `pk` of the request
+    """
+
+    @wraps(view_func)
+    async def _wrapped_view(request, *args, **kwargs):
+        try:
+            response = await view_func(request, *args, **kwargs)
+            return response
+        except Http404 as exc:
+            try:
+                redirection = await ClusterRedirection.objects.aget(
+                    deleted_cluster_id=kwargs["pk"]
+                )
+                cluster_id = redirection.cluster_id
+
+                if not cluster_id:
+                    # TODO: should redirect to sealed page
+                    raise exc
+            except ClusterRedirection.DoesNotExist:
+                raise exc
```

Comment:
sorry, here 

                if redirection.reason == ClusterRedirection.SEALED:
                    html = await sync_to_async(render_to_string)(
                        "410.html", request=request
                    )
                    return HttpResponse(html, status=410)

I think something like this could handle the sealed example